commons/src/main/java/org/calrissian/accumulorecipes/commons/iterators/support/EventFields.java

```diff
@@ -72,7 +72,13 @@ public EventFields read(Kryo kryo, Input input, Class<EventFields> eventFieldsCl
         int entries = intSerializer.read(kryo, input, Integer.class);
         for (int i = 0; i < entries; i++) {
             // Read in the key
-            String key = stringSerializer.read(kryo, input, String.class);
+            String key = null;
+            try {
+                key = stringSerializer.read(kryo, input, String.class);
+            } catch (Throwable e) {
+                System.out.println(input);
```

Comment:
The system.out is what helped me find the EventFields.clear() bug.  I pushed it to make sure the tests still ran.  For some reason the tests failures are different for me on my mac than on travis.  This branch needs to be rebased anyway since it depends on the other issues that you just merged.